dashboard/lib/projects_list.rb

```diff
@@ -140,13 +140,22 @@ def include_featured(limit:)
     #   PUBLISHED_PROJECT_TYPE_GROUPS.keys, or 'all' to retrieve all project groups.
     # @param featured_before [string] String representing a DateTime before
     #   which to search for the requested featured projects. Optional.
-    # @return [Hash<Array<Hash>>] A hash of lists of active published feature projects.
+    # @return [Hash<Array<Hash>>] A hash of lists of active published featured projects.
     def fetch_active_published_featured_projects(project_group, featured_before: nil)
       if project_group == 'all'
         return fetch_featured_published_projects(featured_before: featured_before)
       end
       raise ArgumentError, "invalid project type: #{project_group}" unless PUBLISHED_PROJECT_TYPE_GROUPS.key?(project_group.to_sym)
-      fetch_featured_projects_by_type([project_group.to_sym], featured_before: featured_before)
```

Comment:
`fetch_acitve_published_featured_projects` was added by https://github.com/code-dot-org/code-dot-org/pull/56666 and at the time, we updated the public gallery to display only featured projects and no longer grouped them by type.
Refactored this part of the function so that we can now successfully fetch active, published featured projects by project_group, e.g., 'music'.

Reply:
Not sure I'm totally following, sorry...can you elaborate on why we can't just call `fetch_featured_projects_by_type('music')` directly in the controller rather than having to do this refactor?

Reply:
Sure! so a `project_group` such as 'artist' could have more than one standalone app associated with it. For 'artist' could be either 'frozen' or 'artist' or 'artist_k1' `project_type`. https://github.com/code-dot-org/code-dot-org/blob/77648121f1c76a83dbf95aa7e470a6a9d8f7843d/dashboard/app/models/levels/artist.rb#L49-L57
`PROJECT_TYPE_GROUPS` defined at https://github.com/code-dot-org/code-dot-org/blob/77648121f1c76a83dbf95aa7e470a6a9d8f7843d/dashboard/lib/projects_list.rb#L11.

`fetch_featured_projects_by_type` expects `project_type` as its param. Also - I wanted the return to be a hash so it has the same type as when the project_group is assigned 'all'. Had to update a unit test due to this change at https://github.com/code-dot-org/code-dot-org/blob/f3afd9e56968cb3b871066ea7e0756c8c039dc9c/dashboard/test/controllers/api/v1/projects/public_gallery_controller_test.rb#L17.